lib/truffle/ffi/library.rb

```diff
@@ -50,7 +50,7 @@ def self.map_library_name(lib)
 
     if lib && File.basename(lib) == lib
       lib = Platform::LIBPREFIX + lib unless lib =~ /^#{Platform::LIBPREFIX}/
-      r = Platform::IS_GNU ? "\\.so($|\\.[1234567890]+)" : "\\.#{Platform::LIBSUFFIX}$"
+      r = Platform::IS_GNU ? '\\.so($|\\.[1234567890]+)' : "\\.#{Platform::LIBSUFFIX}$"
```

Comment:
This is an example where it's silly, both sides of the conditional have to use different quoting styles.